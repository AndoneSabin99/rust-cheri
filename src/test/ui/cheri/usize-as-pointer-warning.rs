#![feature(bench_black_box)]
#![deny(usize_as_pointer)]

type WrappedPointer = *const u32;

fn main() {
    let value: usize = 195924;
    let _pointer = value as *const u32; //~ ERROR casting an integer to a pointer will produce an invalid pointer on CHERI
    let _null = 0 as *const u32; // allowed.
    let _singalling = 3 as *const u32; // allowed.
    let _pointer = if std::hint::black_box(false) { value } else { 0 } as WrappedPointer; //~ ERROR casting an integer to a pointer will produce an invalid pointer on CHERI
}
