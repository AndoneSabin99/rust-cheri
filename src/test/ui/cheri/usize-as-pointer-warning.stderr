error: casting an integer to a pointer will produce an invalid pointer on CHERI, and will result in crashes if it is ever dereferenced
  --> $DIR/usize-as-pointer-warning.rs:8:17
   |
LL |     let _pointer = value as *const u32;
   |                    ^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/usize-as-pointer-warning.rs:2:9
   |
LL | #![deny(usize_as_pointer)]
   |         ^^^^^^^^^^^^^^^^

error: casting an integer to a pointer will produce an invalid pointer on CHERI, and will result in crashes if it is ever dereferenced
  --> $DIR/usize-as-pointer-warning.rs:11:17
   |
LL |     let _pointer = if std::hint::black_box(false) { value } else { 0 } as WrappedPointer;
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

